lib/_http_outgoing.js

```diff
@@ -138,15 +138,15 @@ OutgoingMessage.prototype._writeRaw = function(data, encoding, callback) {
       this.connection.writable &&
       !this.connection.destroyed) {
     // There might be pending data in the this.output buffer.
-    while (this.output.length) {
-      if (!this.connection.writable) {
-        this._buffer(data, encoding, callback);
-        return false;
+    var outputLength = this.output.length;
+    if (outputLength) {
+      for (var i = 0; i < outputLength; i++) {
+        var cb = this.outputCallbacks[i];
+        this.connection.write(this.output[i], this.outputEncodings[i], cb);
       }
-      var c = this.output.shift();
-      var e = this.outputEncodings.shift();
-      var cb = this.outputCallbacks.shift();
-      this.connection.write(c, e, cb);
+      this.output = [];
```

Comment:
I can see it going either way.  `.length` is a (very magic!) setter in V8; the `array.length = 0` trick might well be slower than creating a new empty array, even if the latter produces a little more garbage.

Using `array.pop()` here could help as well, it's normally faster than `array.shift()` because it needs to do less work.